### PR TITLE
[pigeon] Fix Example from Pigeon + Update Readme

### PR DESCRIPTION
**Pull request by 4rthurmonteiro**
*Replace this paragraph with a description of what this PR is changing or adding, and why. Consider including before/after screenshots.*

*List which issues are fixed by this PR. You must list at least one issue.*

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/pigeon/example/app/ios/Runner/AppDelegate.swift [hunk 1]
```diff
@@ -31,24 +31,6 @@ private class PigeonApiImplementation: ExampleHostApi {
...
-  func callFlutterMethod(
-    aString aStringArg: String?, completion: @escaping (Result<String, Error>) -> Void
-  ) {
-    flutterAPI.flutterMethod(aString: aStringArg) {
-      completion(.success($0))
-    }
-  }
-}
-// #enddocregion swift-class-flutter
-
```
Comment (tarrinneal) [inline packages/pigeon/example/app/ios/Runner/AppDelegate.swift:51]: This code is here to provide example for our README, it shouldn't be removed.

**Review by tarrinneal — CHANGES_REQUESTED** [review]
The "unnecessary code" serves a purpose elsewhere in our codebase, you should revert it's removal.

**Comment by tarrinneal** [comment]
> _Replace this paragraph with a description of what this PR is changing or adding, and why. Consider including before/after screenshots._
> 
> _List which issues are fixed by this PR. You must list at least one issue._
> 
> _If you had to change anything in the [flutter/tests](https://github.com/flutter/tests) repo, include a link to the migration guide as per the [breaking change policy](https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes)._

Also, please follow the instructions provided here.

**Comment by vashworth** [comment]
@4rthurmonteiro Are you still working on this?

**Comment by stuartmorgan** [comment]
Thank you for your contribution. I'm going to close this PR for now since there are outstanding comments, just to get this off our PR review queue. Please don't hesitate to submit a new PR if you have the time to address the review comments. Thanks!

**Pull request closed without merging**